File: src/Student/Student.java
package Student;

public class Student {
    String id;
    String name;
    int english;
    int math;

    public Student(String name, int english, int math) {
        this.name = name;
        this.english = english;
        this.math = math;
    }

    public int highest(){
        int max;
        return max = (english>math)?english:math;

    }

    public void print(){
        int average = getAverage();
        char grading='F';
        System.out.println(name + "\t" + english+ "\t"+ math+ "\t"+ getAverage()+"\t"+ ((average>60)?"PASS":"FAILED")+"\t"+grading);
        switch (average/10){
            case 10:
            case 9:
                grading='A';
                break;
            case 8:
                grading='B';
                break;
            case 7:
                grading='C';
                break;
            case 6:
                grading='D';
                break;
            default:
                grading='F';
        }

//        if (getAverage()>60){
//            System.out.println("\tPASS");
//        }else {
//            System.out.println("\tFAILED");}
    }
    public int getAverage(){
        return (english+math)/2;
    }
}

File: src/Kotlin/Student.kt
package Kotlin

import java.util.*
import kotlin.math.max

fun main() {
    userInput()

//    val stu = Student("Vix",70,90)
//    stu.print()
//    println("highest: ${stu.highest()}")
}

class Student(var name: String, var english: Int, var math: Int) {
    fun print() {
        print("$name\t$english\t$math\t${getAverage()}\t${grading()}\t${highest()}" )
//        println("最高分: ${highest()}")
        print("\t"+grading())
    }
    fun getAverage()= (english + math) / 2
    fun passOrFailed()= if (getAverage()>60) "PASS" else "FAILED"

    fun grading() = when(getAverage()){
        in 90..100 -> 'A'
        in 80.. 89 -> 'B'
        in 70.. 79 -> 'C'
        in 60.. 69 -> 'D'
        else -> 'F'}


    fun highest() = if (english > math) {
        print("English:")
        english
    } else {
        print("English:")
        math
    }

}
private fun userInput() {
    val scanner = Scanner(System.`in`)
    print("enter student's name:")
    var name = scanner.next()
    print("enter student's english score:")
    var english = scanner.nextInt()
    print("enter student's math score:")
    var math = scanner.nextInt()
    val stu = Student(name, english, math)
    stu.print()
}